### ISSUE TITLE
Centos image issues in mac

### ISSUE THREAD

**Issue by shyam334** [GoogleCloudPlatform/container-diff#168]
Hello Guys,

I'm using `container-diff` in mac, with docker-for-mac. 
I mainly use the analyze feature, And It works fine with debian/ubuntu based images.
However with centos based images, I have to run `container-diff` with `sudo` to make it work. Is this expected.

centos 7 without sudo 

```
> container-diff-darwin-amd64 analyze daemon://centos:7 --type=rpm
Retrieving image centos:7 from source Local Daemon
ERRO[0008] Error creating file /var/folders/_q/9t32sq_d1sg5vfv91dw5t0940000gq/T/centos7296082428/root/.bash_logout open /var/folders/_q/9t32sq_d1sg5vfv91dw5t0940000gq/T/centos7296082428/root/.bash_logout: permission denied 
ERRO[0008] Failed to untar layer with error: open /var/folders/_q/9t32sq_d1sg5vfv91dw5t0940000gq/T/centos7296082428/root/.bash_logout: permission denied 
Retrieving analyses

-----RPM-----

Packages found in centos:7: None

```

centos 7 with sudo

```
> sudo container-diff-darwin-amd64 analyze daemon://centos:7 --type=rpm
Retrieving image centos:7 from source Local Daemon
Retrieving analyses

-----RPM-----

Packages found in centos:7:
NAME                                VERSION          SIZE
-acl                                2.2.51           196.4K
-audit-libs                         2.7.6            242.2K
-basesystem                         10.0             0
-bash                               4.2.46           3.5M
...
[snip]
...
```

ubuntu 16.04 without sudo 
```
> container-diff-darwin-amd64 analyze daemon://ubuntu:16.04
Retrieving image ubuntu:16.04 from source Local Daemon
Retrieving analyses

-----Apt-----

Packages found in ubuntu:16.04:
NAME                        VERSION                              SIZE
-adduser                    3.113 nmu3ubuntu4                    648K
-apt                        1.2.24                               3.3M
...
[snip]
...
```

centos image is from the docker library (iid : 3fa822599e10)

I'm running this on macbook (El capitan) on container-diff v0.6.1, with docker-for-mac.

```
> sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G31

> container-diff-darwin-amd64 version
v0.6.1

> docker version
Client:
 Version:      17.09.0-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:40:09 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.09.0-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   afdb6d4
 Built:        Tue Sep 26 22:45:38 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

**Comment by nkubala** [comment]
I'm able to reproduce this on Linux as well, looks like a bug in the file permissions when unpacking the image tar. I'm looking into it

**Comment by nkubala** [comment]
@shyam334 this should be fixed by #210. See the comment I made in there, unfortunately since we're writing to disk any time we set permissions beyond the scope of the user we can't clean up after ourselves, so this will leave behind an image artifact on your machine unless you run as root. Not great, but still better than failing like it did before. This will be out in the next release, sometime in the next few days.
